test/inductor/test_cutlass_backend.py

```diff
@@ -446,35 +446,97 @@ def test_max_autotune_cutlass_backend_regular_mm(
-            for (M, N, K) in shapes
-        ]
-        dynamic_shapes = (
-            {
-                "a": {0: Dim.DYNAMIC, 1: Dim.DYNAMIC},
-                "b": {0: Dim.DYNAMIC, 1: Dim.DYNAMIC},
+
+        # M, N, K
+        shapes = shapes if dynamic else shapes[0:1]
+
+        if dtype == torch.float8_e4m3fn:
+            inputs = []
+            for shape in shapes:
+                M, N, K = shape
+                output_dtype = torch.bfloat16
+                device = "cuda"
+
+                x = torch.randn(M, K, dtype=output_dtype, device=device)
+                w = torch.randn(N, K, dtype=output_dtype, device=device)
+
+                # quantize weight (prior to inference)
+                w_fp8, w_inverse_scale = _quantize_rowwise(w, dtype)
+                w_t_fp8 = w_fp8.t()
+                w_inverse_scale = w_inverse_scale.t()  # scale_b should be (1, N)
+
+                # quantize input x
+                x_fp8, x_inverse_scale = _quantize_rowwise(x, dtype)
+
+                inputs.append((x_fp8, x_inverse_scale, w_t_fp8, w_inverse_scale))
+
+            class MyModel(torch.nn.Module):
+                def forward(self, x_fp8, x_inverse_scale, w_t_fp8, w_inverse_scale):
+                    y = torch._scaled_mm(
+                        x_fp8,
+                        w_t_fp8,
+                        x_inverse_scale,
+                        w_inverse_scale,
+                        None,
+                        out_dtype=torch.bfloat16,
+                        use_fast_accum=False,
+                    )
+                    return y
+
+            dynamic_shapes = (
+                {
+                    "x_fp8": {0: Dim.DYNAMIC, 1: Dim.DYNAMIC},
+                    "x_inverse_scale": {0: Dim.DYNAMIC, 1: 1},
+                    "w_t_fp8": {0: Dim.DYNAMIC, 1: Dim.DYNAMIC},
+                    "w_inverse_scale": {0: 1, 1: Dim.DYNAMIC},
+                }
+                if dynamic
+                else None
+            )
+            model = MyModel().cuda()
+
+            rtol = 1e-2
+            atol = 0.05
+            additional_patches = {
+                "benchmark_epilogue_fusion": False,  # EVT doesn't support benchmark fusion yet
+                "cuda.cutlass_tma_only": True,
             }
-            if dynamic
-            else None
-        )
+        else:
+
+            class MyModel(torch.nn.Module):
+                def forward(self, a, b):
+                    return a @ b
+
+            model = MyModel().cuda()
+
+            inputs = [
+                (torch.randn(M, K).cuda().to(dtype), torch.randn(K, N).cuda().to(dtype))
+                for (M, N, K) in shapes
+            ]
+
+            dynamic_shapes = (
+                {
+                    "a": {0: Dim.DYNAMIC, 1: Dim.DYNAMIC},
+                    "b": {0: Dim.DYNAMIC, 1: Dim.DYNAMIC},
+                }
+                if dynamic
+                else None
+            )
+            rtol = None
+            atol = None
```

Comment:
Could you add a test where we vary the shape in invocation, not just have a dynamic compiel time shape?

Reply:
Wdym? when we don't use AOTI we run with regular torch.compile w/ dynamic=True which is the same thing right?